corehq/apps/events/models.py

```diff
@@ -264,6 +261,7 @@ def mark_attendance(self, attendee_cases, attended_datetime):
         # Creates ATTENDEE_DATE_CASE_TYPE extension cases for this event
         #   and the given attendee_cases. Also sets the
         #   ATTENDED_DATE_CASE_PROPERTY property to given attended_datetime
+        self.get_attended_attendees.clear(self)
         attendee_case_ids = (c if isinstance(c, str) else c.case_id
```

Comment:
Is there a specific reason why this is being done twice (here and on line 244)? :thinking:

Reply:
Are you suggesting that we move it into its own function? ... Maybe like
```python
    def mark_attendance(self, attendee_cases, attended_datetime):
        # ...
        for case_id in iter_case_ids(attendee_cases):
            # ...

def iter_case_ids(cases_or_case_ids):
    for c in cases_or_case_ids:
        yield c if isinstance(c, str) else c.case_id
```

Reply:
Ah, I only see now that these are done in two different functions! Still, I think it might be nice to have it as a helper function instead to keep things DRY and readable.

Reply:
9764d20f287